iceoryx_posh/test/moduletests/test_mepoo_memory_manager.cpp

```diff
@@ -341,13 +322,224 @@ TEST_F(MemoryManager_test, freeChunkMultiMemPoolFullToEmptyToFull)
     EXPECT_THAT(sut->getMemPoolInfo(3).m_usedChunks, Eq(ChunkCount));
 }
 
-TEST_F(MemoryManager_test, getChunkWithSizeZeroShouldFail)
+TEST_F(MemoryManager_test, getChunkWithSizeZeroShouldNotFail)
 {
-    EXPECT_DEATH({ sut->getChunk(0); }, ".*");
+    mempoolconf.addMemPool({32, 10});
+    sut->configureMemoryManager(mempoolconf, allocator, allocator);
+    EXPECT_THAT(sut->getChunk(0U, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT), Eq(true));
 }
 
 TEST_F(MemoryManager_test, addMemPoolWithChunkCountZeroShouldFail)
 {
     mempoolconf.addMemPool({32, 0});
     EXPECT_DEATH({ sut->configureMemoryManager(mempoolconf, allocator, allocator); }, ".*");
 }
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithAllZeroAsParameterResultsInSizeOfChunkHeader)
+{
+    constexpr uint32_t PAYLOAD_SIZE{0U};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{0U};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{0U};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{0U};
+
+    constexpr uint32_t EXPECTED_SIZE{sizeof(iox::mepoo::ChunkHeader)};
+
+    auto chunkSize = iox::mepoo::MemoryManager::requiredChunkSize(
+        PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+
+    EXPECT_THAT(chunkSize, Eq(EXPECTED_SIZE));
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithZeroPayloadAndDefaultValuesResultsInSizeOfChunkHeader)
+{
+    constexpr uint32_t PAYLOAD_SIZE{0U};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT};
+    constexpr uint32_t CUSTOM_HEADER_SIZE{iox::CHUNK_NO_CUSTOM_HEADER_SIZE};
+    constexpr uint32_t CUSTOM_HEADER_ALIGNMENT{iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT};
+
+    constexpr uint32_t EXPECTED_SIZE{sizeof(iox::mepoo::ChunkHeader)};
+
+    auto chunkSize = iox::mepoo::MemoryManager::requiredChunkSize(
+        PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+
+    EXPECT_THAT(chunkSize, Eq(EXPECTED_SIZE));
+}
+
+TEST_F(MemoryManager_test, requiredChunkSizeWithPayloadAlignmentLessThanChunkHeaderAlignmentResultsInAdjacentPayload)
+{
+    constexpr uint32_t PAYLOAD_SIZE{42U};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{alignof(iox::mepoo::ChunkHeader) / 2};
```

Comment:
You are testing here always with an alignment which is power of 2 but actually alignments like 29 and 31 (fun fact: prime twins) are also valid I think. Could you please add test for those lovely prime twins?

Reply:
nope, alignments must always be power of 2

Reply:
Or we adjust the API a bit. That you do not provide the alignment directly as number, you provide which power of 2 your alignment has. Then this payload alignment could also be an uint8_t and we wouldn't require an assert which would mean less error handling, less code to test.

Reply:
well, that would not be idiomatic. Everywhere else you can just use `alignof(T)` but in this case it would be `sqrt(alignof(T))`. I would hate the developer who enforces such an API on me. We could create a `cxx::align_t` and use that instead of the `uint32_t`. What do you think?

Reply:
`cxx::align_t` sounds great!

Reply:
I've come to the conclusion to create `ChunkSettings` struct with those 4 parameter and also move the `requiredChunkSize` calculation to that struct. All the method would expect this struct instead of the 4 parameters. This makes it typesafe and hard to pass different arguments to the `ChunkHeader` ctor. Would like to do it in a follow up PR, since this PR is already quite large and currently this is only used internally